docs/local-development-environment.adoc

```diff
@@ -0,0 +1,182 @@
+= Local Development Environment
+
+You can easily test the playbook in a local Virtual Machine that we
+can start and stop quickly, called a _devbox_. This document lays out
+the details on how to setup such devbox.
+
+== Prerequisites
+
+To be able to run the playbook on a devbox several groups of
+requirements needs to be covered.
+
+=== Image preparation requirements
+
+Trento's Ansible playbook requires target systems with specific
+conditions that are not trivial to provide such as:
+
+- The target of the playbook should be SLES4SAP system.
+- The system needs to already be registered with SCC.
+
+To help out with these requirements, there are scripts and auxiliary
+helper files prepared in `devbox` directory. The scripts are
+POSIX-compliant and expect only widely-available tools. The only tool
+to note explicitly is `envsubst` command from `gettext-runtime`
+package. Nevertheless, it should be available in most base-systems.
+
+=== VM execution requirements
+
+Currently, development boxes are based on Vagrant by using its
+_libvirt_ provider. Thus, requirements for executing the VM:
+
+- `qemu`
+- `libvirt`
+- `vagrant-libvirt`
+- `vagrant`
+
+
+=== System configuration requirements
+
+The devbox assumes that you have `trento.local` as `localhost` alias in your
+`/etc/hosts`.
+
+== Prepare a SLES4SAP base-image
+
+Vagrant needs a base image to start off its VM. This base image is in
+its own format, called _box_. It's a tar-ball of a virtual disk and
+some vagrant-specific metadata. The virtual disk could be in one of
+several supported formats, depending on the virtualization provider to
+be used. The devbox will be using `qemu`, which is supported through
+Vagrant's `libvirt` provider.
+
+Vagrant usually hosts ready-to-use base-boxes of most OSes but there
+are no ready-made SLES4SAP images available. SUSE only provides SLES
+KVM/Xen images, mainly for use in OpenStack. Thus, for creating
+appropriate Vagrant `box` the following steps should be conducted:
+
+- Convert SLES KVM/Xen image to a box format.
+- Specialize the SLES base-box into a SLES4SAP VM appliance during first boot.
+
+=== SLES Vagrant base-box
+
+To create a SLES base-box, do the following steps:
+
+[NOTE]
+====
+These steps need to be executed only once for a SLES image of specific
+version. Eg. once for SLES15-SP6 and again when trying with SLES15-SP7.
+====
+
+. Download a desired SLES KVM/Xen image from `download.suse.com` or
+from https://scc.suse.com[SUSE Customer Center] (you'd need to have a
+customer account for the later). Again, there is no SLES4SAP VM images
+already prepared from SUSE. The good thing, though, is that both
```

Comment:
You could shorthen up the sentence The good thing, though, is that both systems share the same base installation--> Both systems share the same base installation